ndc-reference/bin/reference/main.rs

```diff
@@ -1833,10 +1870,37 @@ fn eval_nested_field(
                     }),
                 )
             })?;
-            let result_array = array
+            let mut limit = array.len();
+            let limit_error = (StatusCode::INTERNAL_SERVER_ERROR,
+                                Json(models::ErrorResponse {
+                                    message: "Cannot decode `limit` argument".into(),
+                                    details: serde_json::Value::Null, }));
+            if let Some(map) = arguments {
+                println!("Unevaluated arguments: {:?}", arguments);
+
+                if let Some(unevaluated_argument) = map.get("limit") {
+                    // limit = value;
+                    let evaluated_argument = eval_argument(variables, unevaluated_argument)?;
+                    println!("Evaluated argument: {:?}", evaluated_argument);
+                    match evaluated_argument {
+                        Value::Number(n) => {
```

Comment:
You can use `value.as_i64()` here, there are several examples throughout this file. Can you also please read `limit` once, like we do for other input examples.

E.g.

```rust
let id: i32 = row
            .get("id")
            .ok_or("'id' field not found in json file")?
            .as_i64()
            .ok_or("'id' field was not an integer in json file")?
            .try_into()?;
```

Reply:
Sure thing.

Reply:
Actually I cast as usize since this is used by arrays to for bounds, so the marshalling here should do an extra error check for free. What do you think?

Reply:
`let mut limit = array.len();`